### PR TITLE
Add script 2 for fine annotation

### LINKED ISSUE #112: Proposed Analysis: Add script 2 for fine annotation
## What are the scientific goals of the analysis?

To add script 2 for fine annotation, i.e., `02-cell-types-annotation-SingleR-fine.Rmd`.

### PR DESCRIPTION
**Pull request by AntoniaChroni**
# Purpose/implementation Section

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## What scientific question/module of analysis is your analysis addressing?

To add script 2 for fine annotation, i.e., `02-cell-types-annotation-SingleR-fine.Rmd`.

## What GitHub issue does your pull request address?

Closes #112.

## Directions for reviewers 
Tell potential reviewers what kind of feedback you are soliciting.

### Which areas should receive a particularly close look?

Please review for documentation and code logic and clarity.

## Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.

## Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/cell-types-annotation/02-cell-types-annotation-SingleR-fine.Rmd [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+  mutate(cells_number_all = Freq) %>% 
+  dplyr::select(-Freq) %>%
+  left_join(summary_table_pruned)
+```
+
+We will check and compare the pruned labels and the initial labels by summarizing the identified fine cell type annotations.
+
+```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
+tables1 <- summary_table
+cat("  \n<div align=\"center\", style=\"font-size:80%\">  \n")
```
Comment (sharonfreshour) [inline analyses/cell-types-annotation/02-cell-types-annotation-SingleR-fine.Rmd:233]: I think this should be updated to match `pandoc` bug fix from PR #115 

```suggestion
cat("  \n<div align=\"center\" style=\"font-size:80%\">  \n")
```

File: analyses/cell-types-annotation/02-cell-types-annotation-SingleR-fine.Rmd [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+summary_table <- table(cell_type_fine = seurat_obj@meta.data$labels, Removed = to.remove) %>%
+  as.data.frame() %>% 
+  pivot_wider(names_from = "Removed", values_from = "Freq") 
+```
+
+We will summarize the identified fine cell type annotations to be removed.
+
+```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
+tables1 <- summary_table
+cat("  \n<div align=\"center\", style=\"font-size:80%\">  \n")
```
Comment (sharonfreshour) [inline analyses/cell-types-annotation/02-cell-types-annotation-SingleR-fine.Rmd:279]: I think this should be updated to match `pandoc` bug fix from PR #115 

```suggestion
cat("  \n<div align=\"center\" style=\"font-size:80%\">  \n")
```

File: analyses/cell-types-annotation/02-cell-types-annotation-SingleR-fine.Rmd [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+print(knitr::kable(tables1, caption = "Fine cell type annotations to be removed"))
+cat("  \n</div>  \n")
+cat("  \n\\pagebreak  \n")
+```
+
+Low deltas indicate that the assignment is uncertain, which is especially relevant if the cell’s true label does not exist in the reference. This entire process can be visualized ans inspected using the plotScoreDistribution() function, which displays the per-label distribution of the deltas across cells.
```
Comment (sharonfreshour) [inline analyses/cell-types-annotation/02-cell-types-annotation-SingleR-fine.Rmd:285]:
```suggestion
Low deltas indicate that the assignment is uncertain, which is especially relevant if the cell’s true label does not exist in the reference. This entire process can be visualized and inspected using the plotScoreDistribution() function, which displays the per-label distribution of the deltas across cells.
```

**Review by sharonfreshour — APPROVED** [review]
Code looks good, it is clear and logical.

I made a few suggestions related to table styles and the `pandoc` bug brought up in PR #115 and also corrected one typo in the report text.

**Comment by AntoniaChroni** [comment]
> Code looks good, it is clear and logical.
> 
> I made a few suggestions related to table styles and the `pandoc` bug brought up in PR #115 and also corrected one typo in the report text.

Thanks! Fixed here 45e14f9f14d91e47e7c1269b264b308263e963a0.

**Pull request merged**